### DISCUSSION TITLE
Model Hub or ModelServer?

### DISCUSSION THREAD

**Discussion by metadatadriven** [eclipse-emfcloud/emfcloud#226] [ModelServer]
Hi - looks like the documentation has been updated to reflect a move to a more Typescript-centric architecture (Model Hub and Langium, etc.) but that the codebase has not been updated yet, so the documentation is out of sync.

Am I reading that correctly?

If so, any idea when the codebase will be updated in line with the documentation?

Thanks!

**Comment by eneufeld** [comment]
Hi @metadatadriven ,

Yes, that is correct. We are currently in the process of preparing a contribution for the corresponding code base. Unfortunately, we are waiting for legal reviews an we cannot give a precise ETA. We will answer on this thread once it is available. Sorry for the inconvenience!

**Reply by metadatadriven** [reply]
ok, thanks!

**Reply by metadatadriven** [reply]
Any update on how long the legal review might take?
Thanks in advance!

**Reply by eneufeld** [reply]
Unfortunately the process is slow but we finally have the go ahead.
We hope to open the PR for an IP review in early August and if everything works out we should be able to have a first release in September.

**Reply by luxingxiao** [reply]
I am very interested in the new technical architecture. Will it be released as scheduled in September?

**Reply by eneufeld** [reply]
Unfortunately everything took longer and we didn't manage to open a PR in August as planned.
We are currently preparing the PR.
After the PR is open an IP review will be needed. As soon as it concludes we will create a first release.

**Reply by eneufeld** [reply]
@metadatadriven @luxingxiao with some delay we finally merged the new modelmanager code:
https://github.com/eclipse-emfcloud/modelhub
If something is unclear or doesn't work as expected please open an issue in the corresponding repo.

Feel free to ask here if something is unclear.

**Reply by amiths89** [reply]
@eneufeld I did see Coffee Editor NG in the documentation but not the actual code. Is there any example implementation of the model hub with glsp or how to hook up the java glsp server to the model hub?

**Reply by eneufeld** [reply]
@amiths89 yes unfortunately the coffee editor ng code is not yet ready to be published. We are working on it.
If you are using the java version of the glsp server you will need to use the java based modelserver anyway as it the ts based modelhub won't work with java.
The java solution on the backend is what the current coffee editor is based on.

**Reply by amiths89** [reply]
@eneufeld Right i was trying out the modelserverintegration java example code based on tasklist but looks like it has a format error (https://github.com/eclipse-emfcloud/emfcloud/discussions/234) hence was looking for a more recent example

**Reply by eneufeld** [reply]
We will take a look at the error.
I can't promise anything but we try to have a PR with the first version of the updated coffee editor latest next week.

**Reply by francescomagalini** [reply]
Hi @eneufeld 
any update on this? a model hub enabled coffee editor would help a lot here :)
Thank you for your work

**Reply by eneufeld** [reply]
Hi @francescomagalini ,

Unfortunately I have good and bad news.
The good news is we have an update prepared for the website and in the polishing phase: https://github.com/eclipse-emfcloud/emfcloud-website-source/pull/46

The bad news is that there will not be a coffee editor example with the new model hub.

We observed that ModelHub and Langium each cover a significant portion of the other's feature set. While combining both, as attempted in the Coffee Editor, is possible, it introduces challenges that are not practical to solve.

Instead, we now propose two distinct approaches:

- Using Langium with a Model API on top
- Using ModelHub

Examples for both approaches are available and linked in the PR. Details on when to use which approach can also be found in the PR and will be published on the website.
